ogles_gpgpu/CMakeLists.txt

```diff
@@ -1,29 +1,88 @@
 include(sugar_include)
 sugar_include(.)
 
-add_library(ogles_gpgpu ${OGLES_GPGPU_SRCS} )
+add_library(ogles_gpgpu ${OGLES_GPGPU_SRCS})
 
 if(OGLES_GPGPU_VERBOSE)
   target_compile_definitions(ogles_gpgpu PUBLIC OGLES_GPGPU_VERBOSE=1)
 endif()
 
-if(ANDROID)
+## #################################################################
+## Dependencies - OpenGL stuff
+## #################################################################
+
+if(APPLE)
+  if(IOS)
+    target_link_libraries(ogles_gpgpu PUBLIC
+      "-framework ImageIO"
+      "-framework CoreFoundation"
+      "-framework Foundation" # NSLog
+      "-framework OpenGLES"      
+      )    
+  else()
+    target_link_libraries(ogles_gpgpu PUBLIC
+      "-framework ImageIO"
+      "-framework CoreFoundation"
+      "-framework CoreVideo"
+      "-framework OpenGL"
+      )
+  endif()
+elseif(ANDROID)
+  find_package(Android REQUIRED)
+  find_package(Log REQUIRED)
+  find_package(EGL REQUIRED)
+  find_package(OpenGLES2 REQUIRED)
   target_link_libraries(ogles_gpgpu PUBLIC log android EGL GLESv2)
   target_compile_definitions(ogles_gpgpu PUBLIC EGL_EGLEXT_PROTOTYPES GL_GLEXT_PROTOTYPES)
+else()
+
+  if(OGLES_GPGPU_USE_OSMESA)
+
+    find_library(OSMESA_LIBRARY OSMesa)
+    if(NOT OSMESA_LIBRARY)
+      message(FATAL_ERROR "OGLES_GPGPU was configured to use OSMesa, but the library could not be found")
+    endif()
+
+    get_filename_component(OSMESA_PATH ${OSMESA_LIBRARY} DIRECTORY)
+
+    find_library(GLAPI_LIBRARY glapi HINTS ${OSMESA_PATH} PATHS ${OSMESA_PATH})
+    if(NOT GLAPI_LIBRARY)
+      message(FATAL_ERROR "OGLES_GPGPU was configured to use OSMesa, but the GLAPI dependency was not found")
+    endif()
+
+    find_library(EGL_LIBRARY EGL HINTS ${OSMESA_PATH} PATHS ${OSMESA_PATH})
+    if(NOT EGL_LIBRARY)
+      message(FATAL_ERROR "EGL_LIBRARY was configured to use OSMesa, but the EGL dependency was not found")
+    endif()
+    mark_as_advanced(OSMESA_LIBRARY GLAPI_LIBRARY EGL_LIBRARY)
+
+    target_link_libraries(ogles_gpgpu PUBLIC ${OSMESA_LIBRARY} ${GLAPI_LIBRARY} ${EGL_LIBRARY})
```

Comment:
Good point.  The OSMesa stuff is currently used as a CPU only replacement for OpenGL GPGPU CI tests (currently just Travis, but potentially Appveyor as well) where GPU resources are not available.  This allows testing shader correctness  (GTest, etc) in the cloud.  We would never end up building an `ogles_gpgpu` release with a CPU rendering, so the testing configuration is interfering in some ways with the intended library release build.  The OSMesa lib is using EGL, so I guess we would try to link to a standard system EGL library and see if we can pull in OSMesa + dependencies via `LD_LIBRARY_PATH` at runtime.  This is the first I've encountered OSMesa, so I think this may require further experimentation and testing.   Let me know if you have any thoughts.